js/ui/codeView.js

```diff
@@ -437,12 +449,32 @@ var CodingSession = new Lang.Class({
         this._button = new WindowTrackingButton();
         this._button.connect('clicked', this._switchWindows.bind(this));
 
+        // For now, only makes sense to monitor
+        // hackable state if there is a button
+        this._setupToyAppProxy();
+
         _ensureAfterFirstFrame(actor, () => {
             Main.layoutManager.addChrome(this._button);
             this._synchronizeButton(actor.meta_window);
         });
     },
 
+    _setupToyAppProxy: function() {
+        const ToyAppProxy = Gio.DBusProxy.makeProxyWrapper(ToyAppIface);
+        this._toyAppProxy =
```

Comment:
Are you sure it's harmless to create this proxy for an object that doesn't have that interface?

Reply:
It should be harmless (we should test it), but we should also verify that the property gives us sensible values in that case, since we use its value directly in `_onToyAppPropsChanged()`

Reply:
Also, maybe we should name the interface something else so that it doesn't seem limited to toy apps?

Reply:
Yeah, I tested that case.

If an app doesn't implement the interface, the properties changed signal will never be emitted and the callback never be called, so I don't think we need to worry about the value.

But I can make even more robust lile
```
this._hackable = this._toyAppProxy.Hackable === true;
```

Reply:
> Also, maybe we should name the interface something else so that it doesn't seem limited to toy apps?

Yeah, I thought about this too, lets do it.